listenbrainz/mbid_mapping/mapping/mb_metadata_cache.py

```diff
@@ -396,51 +399,223 @@ def config_postgres_join_limit(self, curs):
         curs.execute('SET from_collapse_limit = 15')
         curs.execute('SET join_collapse_limit = 15')
 
-    def mark_rows_as_dirty(self, recording_mbids: List[uuid.UUID], artist_mbids: List[uuid.UUID], release_mbids: List[uuid.UUID]):
-        """Mark rows as dirty if the row is for a given recording mbid or if it's by a given artist mbid, or is from a given release mbid"""
+    def query_last_updated_items(self, timestamp):
+        # there queries here try to mirror the structure and logic of the main cache building queries
+        # note that the tags queries in any of these omit the count > 0 clause because possible removal
+        # of a tag is also a change.
+        # the last_updated considered and last_updated ignored columns below together list all the last_updated
+        # columns a given CTE touches. any other tables touched by a given CTE do not have a last_updated column.
+
+        # further note that these queries only take updates and deletions into consideration, not deletes
+
+        # 1. artist_rels, artist_data, artist_tags, artist
+        # these CTEs and tables concern artist data and we fetch artist mbids from these. all of the CTEs touch
+        # artist table but do not consider its last_updated column because that is done separately at end. further,
+        # the queries here have been simplified to not include recording tables as that will be considered by a
+        # separate query.
+        #
+        # |   CTE / table   |       purpose                  |  last_updated considered           | last_updated ignored
+        # |   artist_rels   |  artist - url links            |  link relationship related and url | artist
+        # |   artist_data   |  life span, area, type, gender |  area                              | artist
+        # |   artist_tags   |  artist tags                   |  recording_tag, genre              | artist
+        # |   artist        |                                |  artist                            |
+        artist_mbids_query = """
+        WITH artist_mbids(id) AS (
+            SELECT a.id
+              FROM artist a
+              JOIN l_artist_url lau
+                ON lau.entity0 = a.id
+              JOIN url u
+                ON lau.entity1 = u.id
+              JOIN link l
+                ON lau.link = l.id
+              JOIN link_type lt
+                ON l.link_type = lt.id
+             WHERE lt.gid IN ('99429741-f3f6-484b-84f8-23af51991770'
```

Comment:
is there any value to unify this list of linktypes in the update and create query?

Reply:
Done